### ISSUE TITLE
Compile error[E0252]: the name `Deserialize` is defined multiple times

### ISSUE THREAD

**Issue by LittleEntity** [neosmart/securestore-rs#1]
````
error[E0252]: the name `Deserialize` is defined multiple times
 --> /Users/#/.cargo/registry/src/github.com-#/securestore-0.99.1/src/shared.rs:8:20
  |
7 | use serde::{Deserialize, Deserializer, Serializer};
  |             ----------- previous import of the macro `Deserialize` here
8 | use serde_derive::{Deserialize, Serialize};
  |                    ^^^^^^^^^^^--
  |                    |
  |                    `Deserialize` reimported here
  |                    help: remove unnecessary import
  |
  = note: `Deserialize` must be defined only once in the macro namespace of this module

error: aborting due to previous error

For more information about this error, try `rustc --explain E0252`.
error: could not compile `securestore`.
````

https://github.com/neosmart/securestore-rs/blob/e1b52c535062898211a5a3ff049e2e23e876dc90/securestore/src/shared.rs#L7-L8

I fixed this issue temporarily for myself by replacing the lines 
````
use serde::{Deserialize, Deserializer, Serializer};
use serde_derive::{Deserialize, Serialize};
````
with
````
use serde::{Deserialize, Deserializer, Serializer, Serialize};
````

# details
## dependencies in Cargo.toml
````
[dependencies]
rocket = "0.4.2"
anyhow = "1.0.26"
pest = "2.1.2"
pest_derive = "2.1.0"
dotenv = "0.15.0"
securestore = "0.99.1"
lettre = "0.9.2"
lettre_email = "0.9.2"
mime = "0.3.16"

[dependencies.rocket_contrib]
version = "0.4.2"
default-features = false
features = [
    "serve",
    "handlebars_templates",
]

[dependencies.serde]
version = "1.0.104"
default-features = false
features = [
    "derive",
]

[dependencies.diesel]
version = "1.4.3"
default-features = false
features = [
    "sqlite",
]
````

## note
I tried to write as brief as possible without holding relevant information back. Thankyou for your work! I enjoy using it. If more information is needed to fix the issue or to track it down on your dev-environment, let me know.

**Comment by mqudsi** [comment]
Hi there,

Thanks for the bug report. I can't reproduce on my end with a freshly minted `rustc 1.43.0-nightly (8aa9d2014 2020-02-21)` while just building the library itself, but I think the problem is explained via your `cargo.lock` and not your `cargo.toml`, my guess is that `serde_derive` is locked to a different version than that used by the particular `serde` version you have installed. 

Directly using the re-exported macro from serde should be just fine, though!

**Comment by mqudsi** [comment]
Thanks for reporting this and for the pointer on the fix. I've published [SecureStore 0.99.2](https://crates.io/crates/securestore) with the fix in place, if you would like to use that directly.
